internal/pkg/core/plugin/service/app.go

```diff
@@ -274,6 +295,84 @@ func (a *Application) Stop() {
 	a.appLock.Unlock()
 }
 
+func (a *Application) stopAndWatch() {
```

Comment:
Not sure I like this function name, the service may still be running yet marked as stopped by the agent, is that correct?

Reply:
Maybe `attemptStop`?
What do we expect the agent to do in cases where endpoint does not stop as far as it's own control flow?

Reply:
it kind of calls the ```stop```via services apis and it kind of ```watches``` the status, that why the ```stopAndWatch```, but sure, can change.
not sure what is expected if the service never shuts down. the original ticket was that the service never recovers from unhealthy state, when the service is removed or uninstalled. 
guess can keep it in unhealthy state if we still see the service running.